addons/account/models/account_partial_reconcile.py

```diff
@@ -124,6 +124,13 @@ def unlink(self):
         # Remove the matching numbers before reversing the moves to avoid trying to remove the full twice.
         full_to_unlink.unlink()
 
+        # Remove link between invoice(s) and payment(s)
+        all_reconciled_invoice_lines = all_reconciled.filtered(lambda line: line.move_id.is_invoice(include_receipts=True))
+        if all_reconciled_invoice_lines:
+            for line in all_reconciled:
+                if line.move_id.origin_payment_id:
+                    line.move_id.origin_payment_id.invoice_ids = line.move_id.origin_payment_id.invoice_ids.filtered(lambda invoice: invoice not in all_reconciled_invoice_lines.move_id)
```

Comment:
maybe we could use the record set also here instead of a filtered like all_reconciled_invoice_lines.move_id - line.move_id.origin_payment_id.invoice_ids, it would be better in term of performance :smile: